### PR TITLE
🪟 🎨 Update replication page and streams table layout

### PR DESCRIPTION
**Pull request by edmundito**
## What
Resolves https://github.com/airbytehq/airbyte/issues/20969
Replaces https://github.com/airbytehq/airbyte/pull/23065

This change merges the changes in these PRs:

- [x] https://github.com/airbytehq/airbyte/pull/22173
- [x] https://github.com/airbytehq/airbyte/pull/22524
- [x] https://github.com/airbytehq/airbyte/pull/21880

There are 12 different combinations to test these changes:

Pages

* Create connection
* Edit replication settings
* Deleted page replication tables

Apps
* Cloud
* OSS

New table feature
* Enabled
* Disabled

## How
Changes described in each PR. This was created as one feature because each had some conflicts with the other and wanted to introduce the changes to users in one pass.

### CODE REVIEW

File: airbyte-webapp/src/pages/connections/ConnectionReplicationPage/StickyEditControlsContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+import styles from "./StickyEditControlsContainer.module.scss";
+
+export const StickyEditControlsContainer: React.FC<React.PropsWithChildren<unknown>> = ({ children }) => (
```
Comment (krishnaglick) [inline airbyte-webapp/src/pages/connections/ConnectionReplicationPage/StickyEditControlsContainer.tsx:5]: This could be a generic.

Comment (edmundito) [reply airbyte-webapp/src/pages/connections/ConnectionReplicationPage/StickyEditControlsContainer.tsx:5]: @krishnaglick A generic?

File: airbyte-webapp/src/components/connection/ConnectionForm/EditControls.tsx [hunk 2]
```diff
@@ -25,24 +29,23 @@ const EditControls: React.FC<EditControlProps> = ({
   successMessage,
   errorMessage,
   enableControls,
+  hidden,
 }) => {
+  const isButtonDisabled = hidden || isSubmitting || (!dirty && !enableControls);
```
[the remark this answers is not in the document]
Comment (edmundito) [reply airbyte-webapp/src/components/connection/ConnectionForm/EditControls.tsx:34]: Yep, leaving it as it-is since it's the current logic.

**Review by krishnaglick — APPROVED** [review]
CI is passing. LGTM, future work will cover my comments :)

**Comment by octavia-approvington** [comment]
Lets merge it <br/> sheep thumbs up

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**